### PR TITLE
Nightly : Fixed `campaigns/modules/blockwishlist/02*/02*` & `campaigns/functional/BO/14*/12*/02*`

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fixed `campaigns/modules/blockwishlist/02*/02*` & `campaigns/functional/BO/14*/12*/02*`
| Type?             | bug fix
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: 
| UI Tests          | https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/6795686801
| Fixed issue or discussion?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

### CODE REVIEW

**Review by nesrineabdmouleh — DISMISSED** [review]

File: tests/UI/pages/FO/category/index.ts [hunk 1]
```diff
@@ -612,16 +612,21 @@ class Category extends FOBasePage {
...
-    await this.elementVisible(page, this.wishlistModal, 2000);
-    // Click on the first wishlist
-    await page.click(this.wishlistModalListItem);
-    // Wait for the toast
-    await this.elementVisible(page, this.wishlistToast, 2000);
-
-    return this.getTextContent(page, this.wishlistToast);
+    if (!(await this.isAddedToWishlist(page, idxProduct))) {
+      // Click on the heart
+      await page.click(this.productAddToWishlist(idxProduct));
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/category/index.ts]:
```suggestion
      await page.locator(this.productAddToWishlist(idxProduct)).click();
```

File: tests/UI/pages/FO/category/index.ts [hunk 2]
```diff
@@ -612,16 +612,21 @@ class Category extends FOBasePage {
...
+      // Wait for the modal
+      await this.elementVisible(page, this.wishlistModal, 3000);
+      // Click on the first wishlist
+      await page.click(this.wishlistModalListItem);
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/category/index.ts]:
```suggestion
      await page.locator(this.wishlistModalListItem).click();
```

File: tests/UI/pages/FO/category/index.ts [hunk 3]
```diff
@@ -612,16 +612,21 @@ class Category extends FOBasePage {
...
+      // Wait for the toast
+      await this.elementVisible(page, this.wishlistToast, 3000);
+
+      return this.getTextContent(page, this.wishlistToast);
+    }
+
+    // Already added
+    return this.messageAddedToWishlist;
```
Comment (boubkerbribri) [inline tests/UI/pages/FO/category/index.ts:629]: Why returning this ? 
Do you verify it somewhere or just to not return null ?

Comment (Progi1984) [reply tests/UI/pages/FO/category/index.ts:629]: It's verified in the scenario.

**Review by nesrineabdmouleh — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Thanks @Progi1984

**Pull request merged**